### ISSUE TITLE
Excessive, unwanted HTTP requests made to my server by your clients

### ISSUE THREAD

**Issue by tjg1** [LiskHQ/lisk-core#102]
Hello. I'd like to congratulate you on this really neat re-implementation of a "blockchain", or as we, people who actually paid attention in computer science classes like to call them, "Merkle trees," a revolutionary data structure first invented and widely used since at least 1979.

Now, to be perfectly honest with you, I have no idea what one would actually want or need a "blockchain" for (other than tax evasion or mail ordering drugs), but before we get all excited to go get some money out of naive venture capitalists (hi @a16z invest in my tor drug empire when you decide to actually make some money off this bitcoin thing lol), could you explain why my server is receiving hundreds of HTTP requests per second on ports 8000 and 8001, from thousands of random IP addresses seemingly running this software?

My server's IP address is `94.237.41.193`, and in case I didn't make it clear enough, it is not a Lisk node, it's not participating in any Blockchain/Sidechain/Ledger, or any other fairy-tale technology meant to excite CEOs and other non-technical manager types who do not understand the complexity of distributed systems.

I'd like to ask you to either immediately release a hotfix release with my server's IP blacklisted, or pay me $500 US dollars (in real, not imaginary unicorn-fart money) to upgrade my cloud instance to accommodate all this unwanted traffic for the next two years.

### Expected behavior

- My server does not need a Blockchain.
- My server does not want a Blockchain.
- It should not be receiving any traffic from people running this software.

### Actual behavior

- Your software seems to think my server needs a Blockchain, it does not.
- Your software seems to think my server wants a Blockchain, it does not.
- My server is receiving thousands of unwanted HTTP requests on port 8000 and 8001, averaging almost 500MB of unwanted traffic per hour, or 360GB per month.
- It looks like this:

```
GET /socketcluster/?version=1.6.0&protocolVersion=1.0&wsPort=8001&httpPort=8000&nethash=ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511&nonce=1VDK407vm3QIIKK4 HTTP/1.1
Sec-WebSocket-Version: 13
Sec-WebSocket-Key: mFwRx6eI96BFYbwJj6Vy6Q==
Connection: Upgrade
Upgrade: websocket
Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits
```

### Steps to reproduce

- Have a server with the IP address `94.237.41.193`.
- Look at the IPTables logs, wonder why some idiot is trying to DDoS you with HTTP requests on filtered ports, when there are open ports 80 and 443 to hit instead.
- Use tcpdump to capture the HTTP requests. Realise it's actually some poorly written blockchain software. Pretend to act surprised.
- Open a sassy Github issue asking them to either cover the cost of the bandwidth or blacklist my IP address.

### Which version(s) does this affect? (Environment, OS, etc...)

NixOS GNU/Linux. A certified blockchain-free™ environment.

**Comment by karek314** [comment]
Hi,
I'm not from development team. However, let me address some things

> I have no idea what one would actually want or need a "blockchain" for (other than tax evasion or mail ordering drugs)

To begin with, very offensive and ignorant to write something like this here. Maybe you are troll, but I bet you aren't so I take a risk and explain.
Perhaps because it removes third parties, which eventually always get corrupted. Great things are being built on the top of blockchain, if you cared enough to write so elaborate issue on problem which you could have solved with fairly simple IPTABLE rule in fraction of time, maybe you will find some time to research this topic further. Good place to start is learning the intention creator of Bitcoin had.
https://en.bitcoin.it/wiki/Genesis_block
**The Times 03/Jan/2009 Chancellor on brink of second bailout for banks**

Bitcoin with proof of work, economically is very similar to gold, making it very good fit to be stable mean of exchange in future. Since Bretton woods, holding regular currencies is guarantee of loss, bankers, politicians and powerful people created the most elaborate financial scam in the world history. Yet, most people are happy to use it and getting ripped of their wealth as we speak. Holding Bitcoin and crypto is bet against that system, system which is doomed to fail. This is just tip of iceberg, I can go on...

Regarding traffic, probably someone had lisk core node on that IP, peer ip is propagated among other nodes, it will eventually get removed. When, not sure, perhaps Lisk core developer can address that.

If that's big trouble, just add simple IPTABLE rule to drop all connections.

**Comment by tjg1** [comment]
Oh, dear. I don't even know where to start.

A firewall rule does not magically make packets disappear into the void. It accomplishes exactly nothing when my server is getting hammered with 500MB/h of WebSocket handshakes for no reason. I'm already dropping all the connections, but instead of doing the sensible thing and giving up - your software keeps re-establishing the connections and sending more handshakes. Those handshakes keep reaching my server before IPTables ever has a chance to drop them and therefore you are still stealing 350GB of bandwidth from me each month. 

Yet another thing you could have learned by paying attention in those OSI CompSci lectures!

![image](https://user-images.githubusercontent.com/1909894/61388314-9986c400-a8af-11e9-974a-a805fa3c1a97.png)

I always get a good laugh when blockchain-tech evangelists fall back on praising Bitcoin, when somebody challenges them that there is no sensible real-life use case for what they're doing.

I love Bitcoin. It's great. They actually took those Merkle trees I was talking about and came up with a new and unique use case for them. I use it all the time to buy nootropics online. And I actually got into it fairly early and made some good money from it. That's why I don't need to pick up the scraps by being part of the blockchain-tech industry that's filled to the brim with charlatans and dilettantes.

And in the 11 years that Bitcoin has been around, I'm yet to see a single "blockchain technology" start-up come around and manage to solve an actual real-life problem

Now can you make your software stop DDoSing my server? I can use my novel PoS implementation by publishing a massive poop emoji on ports 8080/8081, to prove the IP is mine.

If you still think I'm trolling, I'm actually going to Germany for CCCamp19 next month, I'd be quite happy to meet with a lawyer there to discuss if I have any recourse for legal action under section 303b on Computer Sabotage of the German criminal code.

**Comment by Haaroon** [comment]
According to https://explorer.lisk.io/networkMonitor the IP you have now used to run their software as 'bedlam.megane.space' (?) and is a part of the 'disconnected peers' list. So they think it could become active again one day... Thus its possible its still floating around in one of their peer tables.

**Comment by tjg1** [comment]
Yes, `bedlam.megane.space`, that's the reverse DNS PTR record of my server. You know, DNS. That thing when you type in a domain like `google.com` into your web browser, which lets your computer know what IP address it points to. PTR records are like that, but in reverse. They resolve IP addresses into domain names.

Thing is, I have never run your software on my server. And it will never become active again, because I never intend on running your software on my server. What somebody else did in the past with the same IP address is none of my business. In fact, said PTR record would have never existed while they were doing it, because I'm the one who set it up.

Now let me attempt to get through your thick skulls why what you're doing is bad.

Let's say I have a really bad ISP at home. 1Mbps download. I have a dynamic IP address that changes periodically. I was just assigned a new IP address and it just so happens that whoever had the IP beforehand, was running a Lisk node. Now I'm getting hundreds of unwanted connections per second, that if they don't overload the bottom-of-the-barrel router that my ISP gave me, are going to slow down my Internet connection by 50%. Whose fault is it?

Or maybe in a language you can understand.

Let's say I'm some has-been media personality. I just read about this hot new blockchain tech thing in Fortune magazine. My music career isn't doing so great anymore, so I phone up my loser nerd cousin to try to get in on this thing. She forks bitcoin-core, changes a few strings, and we announce our new ICO on my Twitter account. People fall for it, the money keeps rolling in, so I dump all the profits and get myself a new boat with expensive satellite broadband. Unknowingly to me, the satellite ISP just got a new IPv4 assignment and whoever was using the IP address of my boat previously, was running a Lisk node. It's getting half a GB of traffic per hour that I have to pay $10 / 10MB for. When I get back from my cruise and get the bill, all my ICO profits are gone. Now, whose fault is it?

Now can you please blacklist my IP address from your software? Or even better, fix your software, so it's not DDoSing random IP addresses?

**Comment by handsomematt** [comment]
you are very very smart @tjg1 you must of attended a LOT of computer science classes :)

**Comment by theMadBadger022** [comment]
quite the contrary, @tjg1 IS the Computer Science class we slept through hehe, this is fantastic stuff :D

**Comment by aylaylay** [comment]
@tjg1 beat it, nerd

**Comment by Nazgolze** [comment]
Hello @tjg1. We're sorry for the inconvenience and thank you for reporting the issue. Lisk Core is holding onto disconnected peers for far too long and this will be addressed in a future release.

However, being the decentralized application that it is, there's no guarantee that everybody would upgrade in a timely manner when we release a fix. Indeed, if you look at https://explorer.lisk.io/networkMonitor, you'll see there are over 50 nodes that are not on the latest version. And 1.6.0 has been out since April, so after 3 months, at best, 80% of the network has upgraded. So if we released a fix tomorrow, perhaps 3 months from now, your server might receive 100mb per hour instead of 500mb per hour.

If you really want to stop receiving traffic from Lisk Core nodes, you should change your ip address and update your dns entries accordingly

**Comment by shuse2** [comment]
Closing this issue because it is solved by patch release `v2.0.1`

**Comment by karek314** [comment]
@tjg1 You are either very ignorant or just high level troll. You have minimal understanding of economy behind Bitcoin. 

> no sensible real-life use case for what they're doing.

There are literally thousands of use cases. Removing third party in financial transactions is the biggest revolution in the modern history and that is just beginning. Yet you are so stubborn to educate yourself in that area and admit that. I get that it's hard to take a grasp of situation in which some people get very rich very quickly because price appreciation of Bitcoin, this is just natural wealth transfer mechanism and there are also thousands of people who sold Bitcoin very early and regret that. No need to be bitter, instead forget about your prejudices and read about Bitcoin, its economics and technicals with open mind.
